### PR TITLE
convert wav > m4a

### PR DESCRIPTION
**Pull request by matthewberryman**
don't merge yet I am still setting up some of the back end infrastructure (CF & cert name & DNS)

### CODE REVIEW

File: generate_m4a.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const util = require('util');
+const execFile = util.promisify(require('child_process').execFile);
+const download = require('./common').download;
+const upload = require('./common').upload;
+
+const generateM4A = async (file) => {
+  const outputFile = file.substring(0,file.lastIndexOf('.'))+'.m4a';
+  const { error, stdout, stderr } = await execFile('bin/ffmpeg',['-i', file, '-c:a', 'libfdk_aac', '-b:a', '128k', outputFile]);
```
Comment (mehrdadgit) [inline generate_m4a.js]: does it work as expected on a local machine?

File: master.js [hunk 2]
```diff
@@ -18,10 +18,11 @@ module.exports.start = async (event, context, callback) => {
   let isImage= Boolean(data.ContentType.toLowerCase().match(/image/) || isHEI);
   let isJPEGPNG = Boolean(decodedSrcKey.toLowerCase().match(/(\.png|\.jpg|\.jpeg)$/));
   let isVideo = Boolean(data.ContentType.toLowerCase().match(/video/));
+  let isWav = Boolean(decodedSrcKey.match(/\.wav$/i));
```
Comment (mehrdadgit) [inline master.js]: better to check mime type as per https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#WAVE

Comment (mehrdadgit) [reply master.js]: i.e. ContentType

File: generate_m4a.js (hunk 1 above)
Comment (matthewberryman) [reply generate_m4a.js]: no as I haven't compiled with libfdk_aac
Not sure on the elf binary here; if it doesn't have it on testing I will swap over

File: master.js (hunk 2 above)
Comment (matthewberryman) [reply master.js]: 769a008ad99a063a88613a63f0a2ee33d988f677

**Pull request merged**